queries/jsonnet/highlights.scm

```diff
@@ -30,63 +70,64 @@
   ","
   ";"
   ":"
-  "::"
-  ":::"
 ] @punctuation.delimiter
 
-; Nested field inheritance
-("+" @punctuation.delimiter [":" "::" ":::"])
-
-"=" @operator
-(unaryop) @operator
-(binary
-  operator: (_) @operator)
-
-"for" @repeat
-
-"function" @keyword.function
-
-"in" @keyword.operator
-
 [
- (local)
- (tailstrict)
- "assert"
-] @keyword
+  "::"
+  ":::"
+] @punctuation.special
 
+(field
+  (fieldname) "+" @punctuation.special)
+
+; Imports
 [
-  "else"
-  "if"
-  "then"
-] @conditional
+  (import)
+  (importstr)
+] @include
 
-"error" @exception
+; Fields
 
-[
-  (dollar)
-  (self)
-  (super)
-] @variable.builtin
-((id) @variable.builtin
- (#eq? @variable.builtin "std"))
+(fieldname (id) @field)
+(fieldname (string
+             (string_start) @text.strong
+             (string_content) @field
+             (string_end) @text.strong
```

Comment:
Remove the text.strong here too and just highlight the whole (string) as `@string`

Reply:
In general, the captures have to be purely syntactic; markup is the job of color schemes.

(I.e., do not use any @text captures unless the node contains text markup.)

Reply:
I'd like to agree, I just think displaying it just as `@string` in the fieldname context is just not expressive enough.

Do you know how the color scheme can make distinction between `string_start` and `string_content` so that I can do this kind of markup?

Reply:
Would keeping `(string_content) @field` be an option? This would allow the color scheme to make a distinction.

Reply:
does a string have a different meaning in this context compared to others? if not - then no, highlight the whole string node as `@string`

no other language differentiates the start/end quotes

Reply:
It is not a regular string, it is a key in an object:

```jsonnet
{
  key1: 'value'
  'key2': 'value',
  'functionname'(arg): arg,
}
```

`key2` has a different meaning than `value` but the same meaning as `key1`, but it is still a string so I'd find it useful to highlight the quotes differently from the content.

See also the highlight of the function below, simply displaying that as a string dismisses that the fact that it is a function name.

I'd love suggestions on how we can do this kind of distinction better.

Reply:
okay, my bad, I see what you mean now

you can capture the inner string-content as `@field` or `@label` (what json does) 

if you captured string anywhere above as `@string` the quotes will still be highlighted as such.